src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -1071,46 +1071,21 @@ private BoundExpression MakeLiftedUserDefinedConversionConsequence(BoundCall cal
             TypeSymbol source = rewrittenOperand.Type;
             TypeSymbol target = rewrittenType;
 
-            TypeSymbol t0 = target.StrippedType();
-            TypeSymbol s0 = source.StrippedType();
-
-            if (t0 != target && s0 != source)
-            {
-                // UNDONE: RewriteLiftedIntPtrConversion
-
-                return oldNode != null ?
-                    oldNode.Update(
-                        rewrittenOperand,
-                        conversionKind,
-                        oldNode.ResultKind,
-                        isBaseConversion: oldNode.IsBaseConversion,
-                        symbolOpt: symbolOpt,
-                        @checked: @checked,
-                        explicitCastInCode: explicitCastInCode,
-                        isExtensionMethod: isExtensionMethod,
-                        isArrayIndex: isArrayIndex,
-                        constantValueOpt: constantValueOpt,
-                        type: rewrittenType) :
-                    new BoundConversion(
-                        syntax,
-                        rewrittenOperand,
-                        conversionKind,
-                        LookupResultKind.Viable,
-                        isBaseConversion: false,
-                        symbolOpt: symbolOpt,
-                        @checked: @checked,
-                        explicitCastInCode: explicitCastInCode,
-                        isExtensionMethod: isExtensionMethod,
-                        isArrayIndex: isArrayIndex,
-                        constantValueOpt: constantValueOpt,
-                        type: rewrittenType);
-            }
-
             SpecialMember member = GetIntPtrConversionMethod(source: rewrittenOperand.Type, target: rewrittenType);
             MethodSymbol method = GetSpecialTypeMethod(syntax, member);
             Debug.Assert(!method.ReturnsVoid);
             Debug.Assert(method.ParameterCount == 1);
 
+            if (source.IsNullableType() && target.IsNullableType())
+            {
+                Debug.Assert(target.IsNullableType());
+                return RewriteLiftedUserDefinedConversion(syntax, rewrittenOperand, method, rewrittenType, ConversionKind.IntPtr);
+            }
+            else if (source.IsNullableType())
+            {
+                rewrittenOperand = MakeConversion(rewrittenOperand, source.StrippedType(), @checked);
+            }
```

Comment:
Consider restructuring code as follows in order to evaluate ```source.IsNullableType()``` one time instead of two:
```
if (source.IsNullableType())
{
    if (target.IsNullableType())
    {
        return RewriteLiftedUserDefinedConversion(syntax, rewrittenOperand, method, rewrittenType, ConversionKind.IntPtr);
    }

    rewrittenOperand = MakeConversion(rewrittenOperand, source.StrippedType(), @checked);
}
```